File: odonto_api/app/controllers/product_controller.py
from flask import Blueprint, request, jsonify
from odonto_api.app.service.product_service import ProductService

product = Blueprint('product', __name__)

#Endopoint para obtener todos los productos
@product.route( '/productos', methods=['GET'])
def get_products():
    try:
        producto = ProductService.get_all()
        return jsonify(producto), 200
    except Exception as e:
        return jsonify({'error': str(e)}), 500

# Endpoint para obtener producto filtrado por id
@product.route('/productos/<id>', methods=['GET'])
def get_id(id):
    #id = request.args.get('id')
    try:
        producto = ProductService.get_by_id(id)
        return jsonify(producto), 200
    except Exception as e:
        return jsonify({'error': str(e)}), 404


# Endpoint para obtener producto filtrado por coincicencia en nombre y ordenado por precio ASC
@product.route('/productos/nombre', methods=['GET'])
def get_name():
    nombre = request.args.get('nombre')
    try:
        producto = ProductService.get_by_name(nombre)
        return jsonify(producto), 200
    except Exception as e:
        return jsonify({'error': str(e)}), 500


#Endopoint para obtener producto filtrado por categoria y ordenado por precio ASC
@product.route('/productos/categoria', methods=['GET'])
def get_category():
    categoria = request.args.get('categoria')
    try:
        producto = ProductService.get_by_categoria(categoria)
        return jsonify(producto), 200
    except Exception as e:
        return jsonify({'error': str(e)}), 500

#Endopoint para obtener filtrado por marca y ordenado por precio ASC
@product.route('/productos/marca', methods=['GET'])
def get_marca():
    marca = request.args.get('marca')
    try:
        producto = ProductService.get_by_marca(marca)
        return jsonify(producto), 200
    except Exception as e:
        return jsonify({'error': str(e)}), 500

File: backend_api/src/main/java/com/comparodonto/odontoapi/odonto_api/controller/ProductController.java
package com.comparodonto.odontoapi.odonto_api.controller;


import com.comparodonto.odontoapi.odonto_api.Models.Productos;
import com.comparodonto.odontoapi.odonto_api.service.ProductService;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@CrossOrigin
@RestController
@RequestMapping("/api/odonto_api")
@RequiredArgsConstructor
public class ProductController {

    @Autowired
    private ProductService productService;


    //Endppoint obtener por id
    @GetMapping("/{id}")
    public ResponseEntity<Productos> get_by_id(@PathVariable String id){
        return productService.get_producto_id(id)
                .map(ResponseEntity::ok)
                .orElse(ResponseEntity.notFound().build());
    }
    //Endpoint de obtener por nombre ordenado por precio
    @GetMapping("/nombre")
    public List<Productos>get_by_nombre(@RequestParam String nombre){
        return productService.get_producto_nombre(nombre);
    }

    //Endpoint de obtener por categoria ordenado por precio
    @GetMapping("/categoria")
    public List<Productos>get_by_categoria(@RequestParam String categoria){
        return productService.get_producto_categoria(categoria);
    }

    //Endpoint de obtener por marca ordenado por precio
    @GetMapping("/marca")
    public List<Productos>get_by_marca(@RequestParam String marca){
        return productService.get_producto_marca(marca);
    }

    //Endppoint para obtener todos los productos
    @GetMapping("/productos")
    public ResponseEntity<List<Productos>>get_productos(){
        List<Productos> productos = productService.get_productos();
        return ResponseEntity.ok(productos);
    }


}
